internal/app/pactproxy/proxy.go

```diff
@@ -25,157 +26,146 @@ var supportedMediaTypes = map[string]func([]byte, *url.URL) (requestDocument, er
-	api := api{
-		target:       target,
-		proxy:        httputil.NewSingleHostReverseProxy(target),
-		interactions: &Interactions{},
-		notify:       NewNotify(),
-		delay:        defaultDelay,
-		duration:     defaultDuration,
-	}
-
-	for path, handler := range map[string]func(http.ResponseWriter, *http.Request){
-		"/ready":                     api.readinessHandler,
-		"/interactions/verification": api.proxyPassHandler,
-		"/pact":                      api.proxyPassHandler,
-		"/interactions/constraints":  api.interactionsConstraintsHandler,
-		"/interactions/modifiers":    api.interactionsModifiersHandler,
-		"/session":                   api.sessionHandler,
-		"/interactions":              api.interactionsHandler,
-		"/interactions/wait":         api.interactionsWaitHandler,
-		"/":                          api.indexHandler,
-	} {
-		server.HandleFunc(path, handler)
-	}
-}
-
 type api struct {
 	target       *url.URL
 	proxy        *httputil.ReverseProxy
 	interactions *Interactions
 	notify       *notify
 	delay        time.Duration
 	duration     time.Duration
+	echo.Context
 }
 
-func (a *api) proxyPassHandler(res http.ResponseWriter, req *http.Request) {
-	a.proxy.ServeHTTP(res, req)
+func (a *api) ProxyRequest(c echo.Context) error {
+	a.proxy.ServeHTTP(c.Response(), c.Request())
+	return nil
 }
 
-func (a *api) readinessHandler(res http.ResponseWriter, req *http.Request) {
-	if req.Method != http.MethodGet {
-		res.WriteHeader(http.StatusMethodNotAllowed)
+func StartProxy(e *echo.Echo, target *url.URL) {
+
+	// Create these once at startup, thay are shared by all server threads
+	a := api{
+		target:       target,
+		proxy:        httputil.NewSingleHostReverseProxy(target),
+		interactions: &Interactions{},
+		notify:       NewNotify(),
+		delay:        defaultDelay,
+		duration:     defaultDuration,
 	}
+
+	e.GET("/ready", a.readinessHandler)
+
+	e.Any("/interactions/verification", a.proxyPassHandler)
+	e.Any("/pact", a.proxyPassHandler)
+
+	e.POST("/interactions/constraints", a.interactionsConstraintsHandler)
+	e.POST("/interactions/modifiers", a.interactionsModifiersHandler)
+
+	e.DELETE("/session", a.sessionHandler)
+
+	e.POST("/interactions", a.interactionsPostHandler)
+	e.DELETE("/interactions", a.interactionsDeleteHandler)
+
+	e.GET("/interactions/wait", a.interactionsWaitHandler)
+
+	e.Any("/*", a.indexHandler)
 }
 
-func (a *api) interactionsConstraintsHandler(res http.ResponseWriter, req *http.Request) {
-	constraintBytes, err := ioutil.ReadAll(req.Body)
-	if err != nil {
-		httpresponse.Errorf(res, http.StatusBadRequest, "unable to read constraint. %s", err.Error())
-		return
-	}
+func (a *api) proxyPassHandler(c echo.Context) error {
+	return a.ProxyRequest(c)
+}
+
+func (a *api) readinessHandler(c echo.Context) error {
+	return c.NoContent(http.StatusOK)
+}
 
-	constraint, err := LoadConstraint(constraintBytes)
+func (a *api) interactionsConstraintsHandler(c echo.Context) error {
+	constraint := interactionConstraint{}
```

Comment:
Bind (unmarshal) into structs is another thing I like - just a helper function but works well